ingestion-edge/ingestion_edge/publish.py

```diff
@@ -123,13 +127,11 @@
             method=request.method,
             args=request.query_string,
             remote_addr=request.ip,
-            content_length=request.headers.get("Content-Length"),
-            date=request.headers.get("Date"),
-            dnt=request.headers.get("DNT"),
             host=request.host,
-            user_agent=request.headers.get("User-Agent"),
-            x_forwarded_for=request.headers.get("X-Forwarded-For"),
-            x_pingsender_version=request.headers.get("X-Pingsender-Version"),
+            **{
+                attr: request.headers.get(header)
+                for header, attr in metadata_headers.items()
```

Comment:
i chose to require `header` and `attr` separately so the server doesn't need to execute `header.replace("-","_")` all the time

we could instead modify the decoder to use dashes or modify `config.py` to generate the map from a list, this was just the easiest way to implement it.

Reply:
If we know we always want the attr name to follow this format, then I like the idea of generating this map in `config.py`. Especially if that lets us change the `METADATA_HEADERS` format to be a comma-separated string rather than a full json blob.